scripts/generate_ssl_cert

```diff
@@ -72,10 +76,22 @@ openssl dhparam 512 > dh.pem
 
 popd
 
+tmpcert="${DIR}/tmpcertcombined.pem"
+
 (cat ${DIR}/privkey.rsa; echo ""; cat ${DIR}/signedpubcert.pem; \
- echo ""; cat ${DIR}/dh.pem) > ${FILE}
-chmod 400 ${FILE}
+ echo ""; cat ${DIR}/dh.pem) > "${tmpcert}"
+
+# Set up tmpcert the way we want it, then use mv to ensure that FILE
+# appears as an atomic event.
+chmod 400 "${tmpcert}"
+mv --no-clobber "${tmpcert}" "${FILE}"
+# Now check whether the mv succeeded (since it returns a 0 exit-code even if FILE existed already).
+if [ -e "${tmpcert}" ]; then
+	# It still exists in original location: it has not been moved.
+	echo "File has been created just now by someone else: cannot overwrite";
+	EXITCODE=3
```

Comment:
Now that we have the trap on EXIT, this line could be just `exit 3` and we could do without the `EXITCODE` variable.
I'm not sure that would be any better though.

Reply:
Good idea. I think that's definitely better then having an extra `EXIT` variable.